packages/labs/router/README.md

```diff
@@ -4,4 +4,193 @@ A router for Lit.
+- A `routes.outlet` that renders the current route's render callback
+- A `routes.link()` method to generate URLs to use in `<a>` tags, etc.
+- A `routes.goto()` method for performing a navigation
+
+## API
+
+### Router
+
+A Router is a controller (a subclass of the Routes controller) for use at the top-level of an application. It's main purpose is to set up global `click` and `popstate` event listeners (which should be installed only once on a page). It can optionally contain route definitions as a convenience.
+
+It can be installed with no configuration:
+
+```ts
+class App extends LitElement {
+  private router = new Router(this);
+}
+```
+
+Or contain route configurations:
+
+```ts
+class MyElement extends LitElement {
+  private router = new Router(this, [
+    {path: '/', render: () => html`<h1>Home</h1>`},
+  ]);
+
+  render() {
+    return router.outlet;
+  }
+}
+```
+
+### Routes
+
+Routes is the main interface into the router API. A Routes controller contains route definitions and the templates that each route renders:
+
+```ts
+class MyElement extends LitElement {
+  private routes = new Routes(this, [
+    {path: '/', render: () => html`<h1>Home</h1>`},
+    {path: '/projects', render: () => html`<h1>Projects</h1>`},
+    {path: '/about', render: () => html`<h1>About</h3>`},
+  ]);
+}
+```
+
+The second argument is the route configuration: an array of `RouteConfig` objects.
+
+#### RouteConfig
+
+A RouteConfig contains at the minimum the pattern to match URLs against and a template to render. Names can be provided to reference routes for link generation.
+
+There are two types of RoutConfig: `PathRouteConfig` and `URLPatternRouteConfig`:
+
+`PathRouteConfig` lets you specify the URL pattern as a path string:
+
+```ts
+{name: 'home', path: '/', render: () => html`<h1>Home</h1>`}
+```
+
+```ts
+export interface PathRouteConfig {
+  name?: string | undefined;
+  path: string;
+  render: (params: {[key: string]: string}) => unknown;
+}
+```
+
+`URLPatternRouteConfig` lets you specify the URL pattern as a [`URLPattern`](https://developer.mozilla.org/en-US/docs/Web/API/URLPattern) object:
+
+```ts
+{pattern: new URLPattern({pathname: '/'}), render: () => html`<h1>Home</h1>`}
+```
+
+```ts
+export interface URLPatternRouteConfig {
+  name?: string | undefined;
+  pattern: URLPattern;
+  render: (params: {[key: string]: string}) => unknown;
+}
+```
+
+#### Render callbacks
+
+The render callback is called when the outlet property of the Routes object is accessed. It is passed an object with the parameters extracted from the matching URL.
+
+Example with named parameter:
+
+```ts
+{
+  path: '/profile/:id',
+  render: ({id}) => html`<x-profile .profileId=${id}></x-profile>`
+}
+```
+
+#### Outlets
+
+An outlet is where a routes object renders the currently selected route's template. It can be used anywhere in the host element's template:
+
+```ts
+html`${this.routes.outlet}`;
```

Comment:
```suggestion
html`<main>${this.routes.outlet}</main>`;
```

Since you wouldn't usually need the `html` wrapper unless you also had some markup to wrap around it. Sticking in a `main` might be a common top-level wrapping pattern.